src/browser-pool/webdriver-pool.ts

```diff
@@ -0,0 +1,99 @@
+import type { ChildProcess } from "child_process";
+import { runBrowserDriver, getDriverNameForBrowserName } from "../browser-installer";
+import type { SupportedBrowser, SupportedDriver } from "../browser-installer";
+
+type DriverVersion = string;
+type Port = string;
+type ChildProcessWithStatus = { process: ChildProcess; gridUrl: string; isBusy: boolean };
+export type WdProcess = { gridUrl: string; free: () => void; kill: () => void };
+
+export class WebdriverPool {
```

Comment:
We have N browsers and K webdriverds.
This class handles webdriver processes:
- when we can reuse existing webdriver process, which is free
- when we need to launch new webdriver process

Reply:
`port` in this class acts like process id as it is unique